test/functional/Java9andUp/src/org/openj9/test/java/lang/Test_Math_Fma.java

```diff
@@ -0,0 +1,549 @@
+/*******************************************************************************
+ * Copyright (c) 2020, 2020 IBM Corp. and others
+ *
+ * This program and the accompanying materials are made available under
+ * the terms of the Eclipse Public License 2.0 which accompanies this
+ * distribution and is available at https://www.eclipse.org/legal/epl-2.0/
+ * or the Apache License, Version 2.0 which accompanies this distribution and
+ * is available at https://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * This Source Code may also be made available under the following
+ * Secondary Licenses when the conditions for such availability set
+ * forth in the Eclipse Public License, v. 2.0 are satisfied: GNU
+ * General Public License, version 2 with the GNU Classpath
+ * Exception [1] and GNU General Public License, version 2 with the
+ * OpenJDK Assembly Exception [2].
+ *
+ * [1] https://www.gnu.org/software/classpath/license.html
+ * [2] http://openjdk.java.net/legal/assembly-exception.html
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0 WITH Classpath-exception-2.0 OR LicenseRef-GPL-2.0 WITH Assembly-exception
+ *******************************************************************************/
+package org.openj9.test.java.lang;
+
+import org.testng.annotations.Test;
+import org.testng.AssertJUnit;
+
+@Test(groups={ "level.sanity" }, invocationCount=2)
+public class Test_Math_Fma 
+    {
+    @Test
+	public void test_Math_fma_double() 
+        {
+        /**
+         * Testing special case: 
+         * If any argument is NaN, the result is NaN.
+         */
+        double a, b, c, r;
+        a = Double.NaN;
+        b = Math.random(); 
+        c = Math.random();
+        r = Math.fma(a, b, c);
+        AssertJUnit.assertTrue(Double.isNaN(r));
```

Comment:
I'm a little concerned with the use of `random()` above. If this assert ever fires, do you think it would be useful to know the values of `a`, `b`, and `c` so that we can easily debug and reproduce what happened? We should consider using the `String` argument to `assertTrue` [1] to pass a failure message displaying what the values of the inputs were.

[1] http://junit.sourceforge.net/javadoc/org/junit/Assert.html#assertTrue(java.lang.String,%20boolean)

Reply:
It completely makes sense to display what the random numbers are when the assertion fails. My purpose of randomizing those numbers was to find corner cases that we are unaware of. But in this case, I think it is sufficient enough to just use constant numbers instead.

Reply:
If using random numbers I'd say we need to display the numbers used in the assert failure string. If using constant numbers then we don't need to.